eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -965,8 +965,13 @@ protected function publishUrlAliasesForContent(APIContent $content, $updatePathI
     public function deleteContent(ContentInfo $contentInfo)
     {
         $contentInfo = $this->internalLoadContentInfo($contentInfo->id);
+        $content = $this->internalLoadContent($contentInfo->id);
 
-        if (!$this->repository->canUser('content', 'remove', $contentInfo)) {
+        $canDeleteContent = $this->repository->canUser('content', 'remove', $contentInfo);
+
+        if (!$contentInfo->published && !$this->repository->canUser('content', 'versionremove', $content) && !$canDeleteContent) {
```

Comment:
This should probably be written in a more readable way btw, the joined negatives makes it for some fun reading to follow what we expect here :)

Maybe:
```php
if ($contentInfo->isDraft()) {
    // If content is a draft, also allow removal if user has versionremove accesss
    if (
        !$this->repository->canUser('content', 'remove', $contentInfo) &&
        !$this->repository->canUser('content', 'versionremove', $contentInfo)
    ) {
        throw new UnauthorizedException('content', 'versionremove', array('contentId' => $content->id));
    }
} elseif (!$this->repository->canUser('content', 'remove', $contentInfo)) {
    throw new UnauthorizedException('content', 'remove', array('contentId' => $contentInfo->id));
}
```
I didn't manage to avoid adding comment to describe what is happening here, one way might be to use variables that won't need negative value. Aka:
```php
$canNotDeleteContent = !$this->repository->canUser('content', 'remove', $contentInfo);
$canNotDeleteDraft = !$this->repository->canUser('content', 'versionremove', $contentInfo);
```

And then use those inline.

WDYT @bdunogier ?